src/components/layout/header/scroll-effects.ts

```
@@ -3,58 +3,95 @@
+import type { Destroyable, EventCleanup } from "./types";
+
+export class ScrollEffectsManager implements Destroyable {
+  private headers: HTMLElement[] = [];
+  private scrollThreshold = 20;
+  private ticking = false;
+  private eventCleanups: EventCleanup[] = [];
+  private observer: IntersectionObserver | null = null;
+  private sentinel: HTMLElement | null = null;
+  private useIntersectionObserver = false;
+
+  constructor(threshold: number = 20, useIntersectionObserver: boolean = false) {
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)